packages/type-utils/src/propertyTypes.ts

```diff
@@ -34,3 +34,18 @@ export function getTypeOfPropertyOfType(
     property.getEscapedName(),
   );
 }
+
+// Symbolic names need to be specially handled because TS api is not sufficient for these cases.
```

Comment:
Could you drop a permalink here to where these were copied from in the TypeScript source? It helps explain why this odd-seeming hardcoding is doing here.

Reply:
Hm... permalink is not actually permalink? https://github.com/typescript-eslint/typescript-eslint/pull/4343#discussion_r774919883

This is the updated link
https://github.com/microsoft/TypeScript/blob/0043abe982aae0d35f8df59f9715be6ada758ff7/src/compiler/utilities.ts#L3388-L3402

Reply:
just a reminder since your PR was some time ago. As I understood, this https://github.com/typescript-eslint/typescript-eslint/pull/2567  fixes the normal symbol cases and this PR fixes private identifier symbol case

Reply:
Ha, and I didn't add a permalink myself - whoops.